rust/core-lib/src/editor.rs

```diff
@@ -961,6 +961,20 @@ impl Editor {
         self.view.collapse_selections(&self.text);
     }
 
+    fn transform_text(&mut self, transform_function: &Fn(String) -> String) {
```

Comment:
I think a more idiomatic signature for this function would be:

    fn transform_text<F: Fn(String) -> String>(&mut self, transform_function: F)

I'm not sure about this, and looking at relevant docs. Most examples use this pattern, but perhaps `&Fn` might be appropriate, as it might be smaller compiled code (less duplication on monomorphization). Was wondering your thoughts as you wrote this.

I'm also wondering why the argument is an owned `String` rather than a `&str` ref; the latter might afford an opportunity to optimize, as `slice_to_string` has a TODO to be a `Cow`, which would let it apply the same optimization as the LinesRaw iterator.

Reply:
I was also a bit unsure when I wrote it and looked at documentation and after reading them thought that `&Fn` might be better since otherwise code might be duplicated. However, now that I read the docs again I am not sure if that is actually a problem for this case.

As for `String` vs `&str` I didn't put much thought into that. But now that you are pointing it out, it might be better to use `&str` instead.